workers/tests/unittests/test_ohsome_client.py

```diff
@@ -143,11 +151,162 @@ def test_build_data_with_time(self) -> None:
         data = ohsome_client.build_data_dict(layer, self.bpolys, time="2014-01-01")
         self.assertTrue(schema.is_valid(data))
+
+    def test_query_layer_data_valid_1(self):
+        data = asyncio.run(
+            ohsome_client.query(
+                LayerData(
+                    "name",
+                    "description",
+                    {
+                        "result": [
+                            {"value": 1.0, "timestamp": "2020-03-20T01:30:08.180856"}
+                        ]
+                    },
+                )
+            )
+        )
+        self.assertDictEqual(
+            {"result": [{"value": 1.0, "timestamp": "2020-03-20T01:30:08.180856"}]},
+            data,
+        )
+
+    def test_query_layer_data_valid_2(self):
+        data = asyncio.run(
+            ohsome_client.query(
+                LayerData(
+                    "name",
+                    "description",
+                    {
+                        "result": [
+                            {
+                                "value": 1.0,
+                                "fromTimestamp": "2020-03-20T01:30:08.180856",
+                                "toTimestamp": "2020-03-20T01:30:08.180856",
+                            }
+                        ]
+                    },
+                )
+            )
+        )
+        self.assertDictEqual(
+            {
+                "result": [
+                    {
+                        "value": 1.0,
+                        "fromTimestamp": "2020-03-20T01:30:08.180856",
+                        "toTimestamp": "2020-03-20T01:30:08.180856",
+                    }
+                ]
+            },
+            data,
+        )
+
+    def test_query_layer_data_invalid_empty(self):
+        with self.assertRaises(LayerDataSchemaError):
+            asyncio.run(ohsome_client.query(LayerData("name", "description", {})))
+
+    def test_query_layer_data_invalid_empty_list(self):
+        with self.assertRaises(LayerDataSchemaError):
+            asyncio.run(
+                ohsome_client.query(LayerData("name", "description", {"result": []}))
+            )
+
+    def test_query_layer_data_invalid_missing_key(self):
+        with self.assertRaises(LayerDataSchemaError):
+            asyncio.run(
+                ohsome_client.query(
+                    LayerData("name", "description", {"result": [{"value": 1.0}]})
+                )
+            )
+
+    def test_validate_query_results_valid_1(self):
+        ohsome_client.validate_query_results(
+            {"result": [{"value": 1.0, "timestamp": "2020-03-20T01:30:08.180856"}]}
+        )
+
+    def test_validate_query_results_valid_2(self):
+        ohsome_client.validate_query_results(
+            {
+                "result": [
+                    {
+                        "value": 1.0,
+                        "fromTimestamp": "2020-03-20T01:30:08.180856",
+                        "toTimestamp": "2020-03-20T01:30:08.180856",
```

Comment:
Not 100% sure, but I *think* this range is invalid as it doesn't represent a timespan but only one moment in time.

Reply:
I changed to toTimestamp to be one month ahead in contrast to the fromTimestamp